reference-implementation/to-upstream-wpts/writable-streams/aborting.js

```diff
@@ -43,5 +41,256 @@ promise_test(t => {
     assert_not_equals(writer.ready, readyPromise, 'the ready promise property should change');
     return promise_rejects(t, new TypeError(), writer.ready, 'the ready promise should reject with a TypeError');
   });
-
 }, 'Aborting a WritableStream should cause the writer\'s fulfilled ready promise to reset to a rejected one');
+
+function promise_fulfills(expectedValue, promise, msg) {
+  promise.then(value => {
+    assert_equals(value, expectedValue, msg);
+  }, reason => {
+    throw new Error(msg + ': Rejected unexpectedly with: ' + reason);
+  });
+}
+
+promise_test(() => {
+  const ws = new WritableStream({});
+
+  const writer = ws.getWriter();
+  writer.releaseLock();
+
+  const abortPromise = writer.abort();
+  return abortPromise.then(() => {
+    throw new Error('abortPromise fulfilled unexpectedly');
+  },
+  r => {
+    assert_equals(r.constructor.toString(), TypeError.toString(), 'abort() should reject with a TypeError');
+  });
+}, 'abort() on a released writer rejects');
+
+promise_test(() => {
+  const ws = new WritableStream({
+    write() {
+      throw new Error('Unexpected write() call');
```

Comment:
It's a decent bit of work, but I think these sorts of tests would be improved by using the recordingWritableStream, and then checking the recorded events in a final `.then` block. An example: https://github.com/whatwg/streams/blob/ff1d3c9af8a01f4b03a508281844b72a92b533c7/reference-implementation/to-upstream-wpts/writable-streams/close.js#L65-L81